lib/internal/modules/esm/loader.js

```diff
@@ -168,7 +177,16 @@ class Loader {
     return module.getNamespace();
   }
 
-  hook({ resolve, dynamicInstantiate, getFormat, getSource, transformSource }) {
+  hook(hooks) {
+    const {
+      resolve,
+      dynamicInstantiate,
+      getFormat,
+      getSource,
+      transformSource,
+      getGlobalPreloadCode,
```

Comment:
Should `getGlobalPreloadCode` be first?

Reply:
It could be but right this matches the order from the code below (`resolve` being first). I don't think there's a strong forcing factor here for the order.